### DISCUSSION TITLE
Using wire:confirm and wire:submit together - button stays disabled when confirm is cancelled.

### DISCUSSION THREAD

**Discussion by francoisjohannesdutoit** [livewire/livewire#7176] [1. Help] [answered]
I have this code below.

The confirm prompt opens when the button is clicked. When I click cancel, the button stays disabled.

Expected behaviour: When the confirm is cancelled, the button is re-enabled.

Browser: Chrome
Php: 8.1
Livewire version: 3.0.10
Laravel version: 10.29

`<form class="tw-py-3" wire:submit="refund" wire:confirm="Are you sure you want to refund this transaction?">`
   ` <button type="submit">Refund this Transaction</button>`
`</form>`

**Comment by adriansio** [comment]
Encountered the same bug from the recently added Data Tables screencasts (https://livewire.laravel.com/screencast/data-tables/bulk-actions).

It would be great to find a solution to make `wire:confirm` work well with the form submit.

**Comment by jakewhiteley** [comment]
Same. It's quite annoying that using `wire:confirm` and `wire:confirm.prompt` will add disabled to all inputs/selects within that form if the user doesn't confirm.

I wanted to use `wire:confirm` to make a user confirm their choice, so if they had selected the wrong one they could go back and change it, but this is currently not possible.

I tried using `x-on:click="confirm('...') || event.stopImmediatePropagation()"` and other similar approaches, but the `wire:submit` update would still run regardless of user confirmation.

**Reply by jakewhiteley** [reply]
I found a solution.

Using OPs example:

```html
<form class="tw-py-3" x-on:submit="confirm('Are you sure you want to refund this transaction?') ? (event.preventDefault(), $wire.refund()) : event.preventDefault()">
   <button type="submit">Refund this Transaction</button>
</form>
```

This is clearly a bug though in the source livewire code.

**Reply by joshhanley** [reply]
@jakewhiteley yep sounds like a bug. Can you submit a failing test PR so we can investigate? Thanks!

**Reply by jakewhiteley** [reply, marked as answer]
@joshhanley PR submitted - https://github.com/livewire/livewire/pull/8039

**Reply by joshhanley** [reply]
@jakewhiteley legend! Thanks 🙏

**Reply by adriansio** [reply]
@jakewhiteley thanks so much! Looking forward to have this fixed. It's great to rely on the native behavior of forms in Livewire but they didn't work well with wire:confirm.

What is the current solution for this? To use Alpine?

**Reply by jakewhiteley** [reply]
@adriansio pretty much - I put a workaround in a comment above.

**Comment by i6media** [comment]
Alpine workaround:

`<form x-data @submit.prevent="if (confirm('Are you sure?')) $wire.save()">`

instead of:

` <form wire:submit="save" wire:confirm="Are you sure?">`

**Reply by adriansio** [reply]
Does this keep the native behavior of Livewire to disable the form buttons and prevent double submission?

**Reply by i6media** [reply]
It does
